dewy/document/router.py

```diff
@@ -255,3 +246,23 @@ async def get_document_status(conn: PgConnectionDep, id: PathDocumentId) -> Docu
     return DocumentStatus(
         id=id, ingest_state=result["ingest_state"], ingest_error=result["ingest_error"]
     )
+
+@router.delete("/{id}")
+async def delete_document(pg_pool: PgPoolDep, id: PathDocumentId) -> Document:
+    """Delete a document."""
+
+    async with pg_pool.acquire() as conn:
+        deleted = await conn.execute(
+            """
+            DELETE from document
+            WHERE id = $1
+            RETURNING id
+            """,
+            id,
+        )
+        if not deleted:
```

Comment:
Test the cases of deleting a collection / document that doesn't exist (make sure that works as expected).